cmake/macros/macro_populate_target_properties.cmake

```diff
@@ -126,4 +126,13 @@ function(populate_target_properties _target _build)
     ${DEAL_II_LIBRARIES} ${DEAL_II_LIBRARIES_${_build}}
     ${DEAL_II_TARGETS} ${DEAL_II_TARGETS_${_build}}
     )
+
+
+   # For release builds (and their corresponding object files),
+   # use interprocedural optimizations if possible
+   if (DEAL_II_COMPILER_SUPPORTS_IPO AND ("${_build}" STREQUAL "RELEASE"))
```

Comment:
@tamiko On a different note, you had suggested to use `DEAL_II_USE_LTO` instead of `DEAL_II_COMPILER_SUPPORTS_IPO` here. I could go either way. I left it as is because I think that what matters is whether we figured that that we can and should actually use IPO, not what the user specified on the command line. I could be convinced to use `DEAL_II_USE_LTO` here, but in that case the right thing to do would probably to error out if the user set `DEAL_II_USE_LTO=ON` but the compiler doesn't support things. Thoughts?

Reply:
@bangerth I would really like to see `DEAL_II_COMPILER_SUPPORTS_IPO` removed for two reasons: (a) I don't like the current logic that enabling `DEAL_II_USE_LTO` might not enable LTO at all. (b) I do very much prefer that only one consistently named boolean is responsible for changing the behavior.

So what about
 - removing the `DEAL_II_COMPILER_SUPPORTS_IPO` boolean and changing the logic over to `DEAL_II_USE_LTO`?
 - changing the configure check to simply `FATAL_ERROR`  if `DEAL_II_USE_LTO` is set but we cannot enable LTO.